tools/converter/mule2ballerina/src/main/java/org/wso2/ei/tools/mule2ballerina/visitor/TreeVisitor.java

```diff
@@ -428,12 +443,37 @@ public void visit(VariableRemover variableRemover) {
         ballerinaASTAPI.createVariableRefExpr();
     }
 
+    /**
+     * When a flow reference is called, it might either refer to a sub flow or a private flow. In case of a sub flow,
+     * add the processors that's been referred in it, in the calling resource. But if it's a private flow call the
+     * respective function.
+     *
+     * @param flowReference
+     */
     @Override
     public void visit(FlowReference flowReference) {
+        //Add the sub flow processors also into the calling resource
         if (mRoot.getSubFlowMap() != null && !mRoot.getSubFlowMap().isEmpty()) {
             SubFlow subFlow = mRoot.getSubFlowMap().get(flowReference.getName());
-            for (Processor processor : subFlow.getFlowProcessors()) {
-                processor.accept(this);
+            if (subFlow != null) {
+                ballerinaASTAPI.addComment("//------Consider wrapping the following logic in a function-----");
+                for (Processor processor : subFlow.getFlowProcessors()) {
+                    processor.accept(this);
+                }
+                ballerinaASTAPI.addComment("//-------------------------------------------------------------");
```

Comment:
Could you explain the role of this operation ?

Reply:
With a flow reference, you could have multiple processors inside it. But after the conversion, we are including the processor logic in the calling control itself, because to wrap it in a  separate function converter cannot decide what parameters it would require. That's why the logic is wrapped with a start and end comment, to instruct the user to consider wrapping it in a separate function.